components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -2396,6 +2397,21 @@ private void parseSupportedGrantTypesConfig(OMElement oauthConfigElem) {
                         refreshTokenAllowedGrantTypes.put(grantTypeName, isRefreshAllowed);
                     }
                 }
+
+                // Read the public client allowed grant types from default grant types.
+                OMElement publicClientAllowedElement = supportedGrantTypeElement
+                        .getFirstChildWithName(getQNameWithIdentityNS(ConfigElements.PUBLIC_CLIENT_ALLOWED));
+                String publicClientAllowed = null;
+                if (publicClientAllowedElement != null) {
+                    publicClientAllowed = publicClientAllowedElement.getText();
+                }
+                if (StringUtils.isNotEmpty(publicClientAllowed)) {
```

Comment:
what will happen if we have a grant type with out the value for publicClientAllowed (if publicClientAllowed is empty)?

Reply:
If `PublicClientNotAllowed` property is not present, then `publicClientAllowedElement` will be null and hence, `publicClientAllowed ` will be null as well. So the loop will continue.

If the property is present and no value is provided, `publicClientAllowedElement` will not be null and `publicClientAllowed` will be an empty string (""). So the check in line 2408 will fail and hence the loop will continue.

Reply:
If it is empty or null then the relevant grant type will be added to publicClientSupportedGrantTypes or publicClientNotSupportedGrantTypes right?

Reply:
Yes. Sorry for the misunderstanding. Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2161/commits/049806cbd0006c7e972b2ff7b6418feeda51c4a5